### ISSUE TITLE
Pass LSP labelDetails field to Completion struct on language extensions

### ISSUE THREAD

**Issue by jotaviobiondo** [zed-industries/zed#14278]
### Check for existing issues

- [X] Completed

### Describe the feature

TLDR: It's possible to add the `label_details` field from `lsp::CompletionItem` struct to the `wit::Completion` struct?

### Problem
Currently, the autocompletion for Elixir is not looking so nice (as it is in Rust and other languages) because it doesn't show the arguments of the function, just the name. This is the completions for Elixir using the `elixir-ls` language server:

![image](https://github.com/user-attachments/assets/843c6cc3-efc6-4049-8398-3ff750df4fbe)

It's hard to identify a function when there's many with the same name but with different arity.

In VSCode we get: ![image](https://github.com/user-attachments/assets/fe09d5be-c84d-4d6d-998a-28ba11bc77e4)

In Zed with rust we get: ![image](https://github.com/user-attachments/assets/e7baff8d-0d95-4926-9c6a-471ac2c2c042)

### Investigation

So I tried to make a PR to implement this for the Elixir language servers, but the main problem is that they (or at least `elixir-ls`) sends the arguments detail inside the `labelDetails` field in the LSP response:

```json
{
  "id":22194,
  "jsonrpc":"2.0",
  "result":{
    "isIncomplete":true,
    "items":[
      {
        "detail":"function",
        "documentation":{...},
        "insertText":"fetch(${1:enumerable}, ${2:index})",
        "insertTextFormat":2,
        "insertTextMode":2,
        "kind":3,
        "label":"fetch",
        "labelDetails":{
            "description":"Enum.fetch/2",
            "detail":"(enumerable, index)"
        },
        "sortText":"00000000"
      },
    ]
  }
}
```

and the `Completion` struct that the language server extensions receive doesn't have this field (just the `detail` one). 
I also checked that for the built in languages the api for languages is a bit different and receive the `CompletionItem` that has the `label_details` field.

I tried to add the new field when converting from the `lsp::CompletionItem` to `wit::Completion` here (crates/extension/src/extension_lsp_adapter.rs):

![image](https://github.com/user-attachments/assets/f8a8e2ae-6257-498e-abc3-5da5541cea6e)

but I couldn't figure out how to modify the `Completion` struct. I tried to modify it in `crates/extension_api/wit/since_v0.0.7/lsp.wit` but it seems that it does not make any difference, maybe I need to recompile this wit files? (I'm not experienced with Rust nor wasm)

So, can we pass the `labelDetails` to the `wit::Completion` struct? And I would also appreciate any help so I can make a PR for it 🙂

And I have a related question but just out of curiosity: Why the language extensions are "behind" this wasm interface? It's related to cross platform?
